### ISSUE TITLE
Docker Hub Rate Limits

### ISSUE THREAD

**Issue by cgreene** [manubot/rootstock#393]
The new Docker Hub rate limits are causing some builds to fail. I especially notice this with appveyor. Here's an example https://ci.appveyor.com/project/cgreene/annorxiver-manuscript/builds/36473656

It might be nice to examine our dependencies and to see which ones are already on the GitHub Container Registry ( https://github.blog/2020-09-01-introducing-github-container-registry/ ).

**Comment by cgreene** [comment]
I agree that it is because of the new Docker Hub limits. I think we might expect to see these becoming pervasive as manubot use increases.

Are you sure that the GitHub Container Registry requires logging in to pull? That link seems to be referring to the GitHub Package Registry. The Container Registry docs say:
> Public images allow anonymous access and can be pulled without authentication or signing in via the CLI.

Also, the naming of this product from GitHub, given the prior one, is just terrible.

**Comment by cgreene** [comment]
pagedjs looks nice too! is there a ticket for that already?

**Comment by dhimmel** [comment]
> Are you sure that the GitHub Container Registry requires logging in to pull? That link seems to be referring to the GitHub Package Registry.

Ah I wasn't aware of the [distinction](https://docs.github.com/en/free-pro-team@latest/packages/getting-started-with-github-container-registry/migrating-to-github-container-registry-for-docker-images) between "GitHub Packages Docker registry" (scoped to a repo) and the GitHub Packages Docker registry (scoped to an account).

That's great that GHCR will allow anonymous access. On the flip side, publishing images to GHCR from GitHub Actions will require custom credential configuration.

> pagedjs looks nice too! is there a ticket for that already?

It comes up in https://github.com/manubot/rootstock/issues/299, but we should have a dedicated issue. Will open.

**Comment by agitter** [comment]
This excerpt from the GitHub page @dhimmel linked above is worth noting:
> Note: GitHub Container Registry is currently in public beta and subject to change. During the beta, storage and bandwidth are free. 

There's a chance that post-beta GitHub Container Registry could have the same limitations as Docker Hub.

**Comment by cgreene** [comment]
Near that it says:
> After the beta, the same billing and storage rates that other GitHub Packages registries use will apply to the container registry. For more information, see "About billing for GitHub Packages."

I do agree there's no strong commitment to unlimited, free pulls, but I think that clause is referring to the changes around private ones.

**Comment by agitter** [comment]
Good catch @cgreene.

The (former) athenapdf maintainer recommended https://thecodingmachine.github.io/gotenberg/ as another alternative.  It looks like it is also distributed as a Docker image.
